### PR TITLE
refactor(date): remove duplicate context of date definitions

### LINKED ISSUE #1577: In many locales, {wide|abbr}_context and {wide|abbr} are the same for faker.date.weekday and faker.date.month
### Pre-Checks

- [X] Follow our [Code of Conduct](https://github.com/faker-js/faker/blob/next/CODE_OF_CONDUCT.md).
- [X] Read the [Contributing Guidelines](https://github.com/faker-js/faker/blob/next/CONTRIBUTING.md).
- [X] Read the [docs](https://fakerjs.dev/guide).
- [X] Check that there isn't [already an issue](https://github.com/faker-js/faker/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a Faker issue and not related to a combination with another package.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/faker-js/faker/discussions) or join our [Discord Chat Server](https://chat.fakerjs.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.
- [X] I am willing to provide a PR.

### Describe the bug

For example in `nl`
https://github.com/faker-js/faker/blob/next/src/locales/nl/date/weekday.ts

the `wide` and `wide_context` arrays are identical. Since `wide` is used automatically if `wide_context` is missing, it can be removed completely from the `nl` locale.

This 
* reduces the code size
* reduces the chance for errors when one array is modified and not the other
* allows focus on the languages which have a genuine need for the context parameter

### Minimal reproduction code

Some quick hacky code to spot duplicates:

```js
const {faker} = require("@faker-js/faker")
Object.keys(faker.locales).forEach(l=>{
    let locale = faker.locales[l]
    if (locale.date && locale.date.weekday) {
        let wc = locale.date.weekday.wide_context
        let w = locale.date.weekday.wide
        if (wc && w && wc.filter(x=>!w.includes(x)).length==0) {
            console.log(l,"weekday, wide")
        }
        let ac = locale.date.weekday.abbr_context
        let a = locale.date.weekday.abbr
        if (ac && a && ac.filter(x=>!a.includes(x)).length==0) {
            console.log(l,"weekday, abbr")
        }
    }
    if (locale.date && locale.date.month) {
        let wc = locale.date.month.wide_context
        let w = locale.date.month.wide
        if (wc && w && wc.filter(x=>!w.includes(x)).length==0) {
            console.log(l, "month, wide")
        }
        let ac = locale.date.month.abbr_context
        let a = locale.date.month.abbr
        if (ac && a && ac.filter(x=>!a.includes(x)).length==0) {
            console.log(l,"month, abbr")
        }
    }
}
```

Output
```
ar weekday, wide
ar month, wide
ar month, abbr
cz weekday, wide
cz weekday, abbr
cz month, wide
cz month, abbr
de weekday, wide
de weekday, abbr
de month, wide
de month, abbr
en weekday, wide
en weekday, abbr
en month, wide
en month, abbr
fa weekday, wide
fa weekday, abbr
fa month, wide
fa month, abbr
fr month, abbr
fr_CH month, abbr
he weekday, wide
he weekday, abbr
he month, wide
he month, abbr
hr weekday, wide
hr weekday, abbr
hr month, wide
hr month, abbr
hu weekday, wide
hu weekday, abbr
hu month, wide
hu month, abbr
id_ID weekday, wide
id_ID weekday, abbr
id_ID month, wide
id_ID month, abbr
mk weekday, wide
mk weekday, abbr
mk month, wide
mk month, abbr
nl weekday, wide
nl weekday, abbr
nl month, wide
nl month, abbr
pt_BR weekday, wide
pt_BR weekday, abbr
pt_BR month, wide
pt_BR month, abbr
ro weekday, wide
ro weekday, abbr
ro month, wide
ro month, abbr
ur weekday, wide
ur month, wide
vi weekday, wide
vi month, wide
vi month, abbr
```

### Additional Context

If you think it makes sense to remove the duplicate wide_context and abbr_context arrays, I can remove them from the specified locales.

### Environment Info

```shell
System:
    OS: macOS 12.6
    CPU: (8) x64 Apple M1
    Memory: 43.10 MB / 8.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 14.20.0 - /usr/local/bin/node
    npm: 8.19.1 - /usr/local/bin/npm
  Browsers:
    Chrome: 107.0.5304.110
    Firefox: 98.0
    Safari: 16.0
```

### Which module system do you use?

- [X] CJS
- [ ] ESM

### Used Package Manager

pnpm

### PR DESCRIPTION
**Pull request by matthewmayer**
fix #1577

Removes wide_context and abbr_context for faker.date.weekday and faker.date.month where they are identical to wide and abbr

As this also includes `en`, i made some changes to a few tests to use an alternative locale `az` for testing the context parameter

### CODE REVIEW

**Review by ST-DDT — DISMISSED** [review]

**Review by Shinigami92 — DISMISSED** [review]

File: src/locales/vi/date/month.ts [hunk 1]
```diff
@@ -13,22 +13,7 @@ export default {
...
-    'Tháng Năm',
-    'Tháng Sáu',
-    'Tháng Bảy',
-    'Tháng Tám',
-    'Tháng Chín',
-    'Tháng Mười',
-    'Tháng Mười Một',
-    'Tháng Mười Hai',
-  ],
+
```
Comment (xDivisionByZerox) [inline src/locales/vi/date/month.ts]: Sorry for being nitpicky, but could we get rid of this empty line?

File: src/locales/vi/date/weekday.ts [hunk 2]
```diff
@@ -8,17 +8,7 @@ export default {
...
-  wide_context: [
-    'Chủ nhật',
-    'Thứ hai',
-    'Thứ ba',
-    'Thứ tư',
-    'Thứ năm',
-    'Thứ sáu',
-    'Thứ bảy',
-  ],
+
```
Comment (xDivisionByZerox) [inline src/locales/vi/date/weekday.ts]: Here as well.

File: src/locales/vi/date/month.ts (hunk 1 above)
Comment (ST-DDT) [reply src/locales/vi/date/month.ts]: Is there a prettier rule for something like that?

Comment (xDivisionByZerox) [reply src/locales/vi/date/month.ts]: Not that I know of. There is an eslint [no-multiple-empty-lines](https://eslint.org/docs/latest/rules/no-multiple-empty-lines) rule, which doesn't apply here, but is the closest that comes t mind.

**Review by ST-DDT — APPROVED** [review]

**Review by xDivisionByZerox — APPROVED** [review]

**Review by Shinigami92 — APPROVED** [review]

**Pull request merged**